modules/spring-config/src/main/java/org/mule/runtime/config/internal/LazyMuleArtifactContext.java

```diff
@@ -179,6 +184,19 @@ private void applyLifecycle(List<String> createdComponentModels, boolean applySt
     });
   }
 
+  private void applyLifecycleMessageProcessorChainBuilder(Object object, String phase) {
+    if (object instanceof MessageProcessorChainBuilder) {
+      MessageProcessorChain messageProcessorChain = ((MessageProcessorChainBuilder) object).build();
+      messageProcessorChain.getMessageProcessors().stream().forEach(processor -> {
```

Comment:
can't you apply lifecycle directly on the chain instead of iteratin and applying on each element?